elastica/rigidbody/mesh_rigid_body.py

```diff
@@ -0,0 +1,198 @@
+
+        self.density = density
+        self.volume = volume
+        self.mass = np.array([self.volume * self.density])
+        normal = np.array([1.0, 0.0, 0.0]).reshape(3, 1)
+        tangents = np.array([0.0, 0.0, 1.0]).reshape(3, 1)
+        binormal = _batch_cross(tangents, normal)
+        self.faces = mesh.faces.copy()
+        self.n_faces = mesh.faces.shape[-1]
+        self.face_centers = np.array(mesh.face_centers.copy())
+        self.face_normals = np.array(mesh.face_normals.copy())
+        self.distance_to_face_centers = _batch_norm(
+            self.face_centers - center_of_mass.reshape(3, 1)
+        )
+        self.direction_to_face_centers = (
+            self.face_centers - center_of_mass.reshape(3, 1)
+        ) / self.distance_to_face_centers
+        self.distance_to_faces = np.zeros((MaxDimension.value(), self.n_faces))
+        self.direction_to_faces = np.zeros(
+            (MaxDimension.value(), MaxDimension.value(), self.n_faces)
+        )
+        for i in range(MaxDimension.value()):
+            for k in range(self.n_faces):
+                self.distance_to_faces[i, k] = np.linalg.norm(
+                    self.faces[:, i, k] - center_of_mass
+                )
+                self.direction_to_faces[:, i, k] = (
+                    self.faces[:, i, k] - center_of_mass
+                ) / self.distance_to_faces[i, k]
+
+        self.mass_second_moment_of_inertia = mass_second_moment_of_inertia.reshape(
+            MaxDimension.value(), MaxDimension.value(), 1
+        )
+
+        self.inv_mass_second_moment_of_inertia = np.linalg.inv(
+            mass_second_moment_of_inertia
+        ).reshape(MaxDimension.value(), MaxDimension.value(), 1)
+
+        # position is at the center
+        self.position_collection = np.zeros((MaxDimension.value(), 1))
+        self.position_collection[:, 0] = center_of_mass
+
+        self.velocity_collection = np.zeros((MaxDimension.value(), 1))
+        self.omega_collection = np.zeros((MaxDimension.value(), 1))
+        self.acceleration_collection = 0.0 * self.velocity_collection
+        self.alpha_collection = 0.0 * self.omega_collection
+
+        self.director_collection = np.zeros(
+            (MaxDimension.value(), MaxDimension.value(), 1)
+        )
+        self.director_collection[0, ...] = normal
+        self.director_collection[1, ...] = binormal
+        self.director_collection[2, ...] = tangents
+
+        test_faces = np.zeros((3, 3, self.n_faces))  # dim,vertices,faces
+        for k in range(self.n_faces):
+            for m in range(3):
+                for i in range(3):
+                    test_faces[i, m, k] += center_of_mass[i]
+                    for j in range(3):
+                        test_faces[i, m, k] += (
+                            self.distance_to_faces[m, k]
+                            * self.director_collection[i, j, 0]
+                            * self.direction_to_faces[j, m, k]
+                        )
```

Comment:
what is this for? Seems to be it is not used outside of the scope of initialization. why do you have an array called test_faces

Reply:
I think I had them to test out something than forgot to remove them, I will remove them